### PR TITLE
Update golang image

### PR DESCRIPTION
**Pull request by antonio-te**
1.17.3 is affected by CVE-2021-44716.

### CODE REVIEW

**Comment by antonio-te** [comment]
> The commit is not authorized under a signed CLA.

I'm an employee of Cisco, which is covered under the corporate CNCF CLA. Any way I can prove that to the bot?

**Review by damemi — COMMENTED** [review]
/approve
/ok-to-test
I think there are other references to 1.17.3, ie in our tests that should probably be updated too

**Comment by damemi** [comment]
> > The commit is not authorized under a signed CLA.
> 
> I'm an employee of Cisco, which is covered under the corporate CNCF CLA. Any way I can prove that to the bot?

@antonio-te if you follow the link in the bot's comment, it will help link your Github account to the CNCF CLA

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM golang:1.17.3
+FROM golang:1.17.6
```
Comment (justaugustus) [inline Dockerfile:14]:
```suggestion
FROM golang:1.17.7
```

ref: https://github.com/kubernetes/release/issues/2425

**Comment by damemi** [comment]
@justaugustus do you know if maybe there is an issue with the CLA bot?

**Comment by justaugustus** [comment]
> @justaugustus do you know if maybe there is an issue with the CLA bot?

This may be something on the internal side. Investigating! :)

**Comment by antonio-te** [comment]
> I think there are other references to 1.17.3, ie in our tests that should probably be updated too

I couldn't find any:
```
~/src/REF/descheduler   master ❯ ag -a '1\.17\.[0-9]+'
Dockerfile
14:FROM golang:1.17.7

go.sum
577:go.uber.org/zap v1.17.0/go.mod h1:MXVU+bhUf/A7Xi2HNOnopQOrmycQ5Ih87HtOu4q5SSo=

vendor/go.uber.org/zap/CHANGELOG.md
47:## 1.17.0 (25 May 2021)

vendor/k8s.io/client-go/tools/cache/reflector.go
597:    // In Kubernetes 1.17.0-1.18.5, the api server doesn't set the error status cause to
610:            // Matches the message returned by api server 1.17.0-1.18.5 for this error condition

vendor/k8s.io/api/core/v1/well_known_labels.go
44:     // LabelWindowsBuild is used on Windows nodes to specify the Windows build number starting with v1.17.0.
~/src/REF/descheduler   master ❯
```

**Comment by damemi** [comment]
@antonio-te sorry, I was more referring to our test images (I opened a PR for that in https://github.com/kubernetes/test-infra/pull/25438). This looks good to me now
/lgtm

**Pull request merged**